test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -760,14 +684,149 @@ private static void CounterUpdateThread<T>(object obj)
             }
         }
 
+        private static void HistogramUpdateThread<T>(object obj)
+            where T : struct, IComparable
```

Comment:
Do we need the constraint for `IComparable`?

Reply:
Removing `IComparable` gives this error
![image](https://user-images.githubusercontent.com/15165028/145309173-00567f9d-253f-4d86-b3c3-234321a5f373.png)

Reply:
Why does `UpdateThreadArguments<T>` need the `IComparable` constraint? We have not overridden any `CompareTo` method for any class in this file.